### PR TITLE
Bug 2119328: ACM plugin custom extension for row action

### PR DESCRIPTION
**Pull request by GowthamShanmugam**
Related to: [Bug 2119328](https://bugzilla.redhat.com/show_bug.cgi?id=2119328)
ODF issue: https://issues.redhat.com/browse/RHSTOR-3486

Signed-off-by: Gowtham Shanmugasundaram <gshanmug@redhat.com>

### CODE REVIEW

File: frontend/src/routes/Applications/Overview.tsx [hunk 1]
```diff
@@ -917,7 +917,7 @@ export default function ApplicationsOverview() {
                 acmExtensions.applicationAction.forEach((appAction) => {
                     if (appAction?.model ? isResourceTypeOf(resource, appAction?.model) : isOCPAppResource(resource)) {
                         const ModalComp = appAction.component
-                        const close = () => setPluginModal(<ModalComp isOpen={false} />)
+                        const close = () => setPluginModal(<></>)
```
Comment (GowthamShanmugam) [inline frontend/src/routes/Applications/Overview.tsx:920]: This change will always open a fresh modal, without this model loading with previous state values.

**Comment by GowthamShanmugam** [comment]
@KevinFCormier @fxiang1

**Comment by fxiang1** [comment]
@GowthamShanmugam Just wondering is this a sev 1 bug? We're at the end of the release so only taking in high severity bugs.

FYI @cbynum-redhat 
FYI @thuyn-581 @asimbutt-redhat in case there's any QE concerns

**Comment by GowthamShanmugam** [comment]
@KevinFCormier @

> @GowthamShanmugam Just wondering is this a sev 1 bug? We're at the end of the release so only taking in high severity bugs.
> 
> FYI @cbynum-redhat FYI @thuyn-581 @asimbutt-redhat in case there's any QE concerns

Kind of bug I displaying actions on the application list page but not on the details page. also fixed one minor bug with the loading modal. 

I want these extension-related changes on ACM 2.6 is enough, ODF 4.12 is going to target ACM 2.6 only

**Comment by GowthamShanmugam** [comment]
image
image

**Comment by GowthamShanmugam** [comment]
@fxiang1 Can we take this for 2.6.Z stream, this is bug fix only?

**Comment by fxiang1** [comment]
@GowthamShanmugam Yes, so once 2.6 is GA @cbynum-redhat will open up the release-2.6 branch for 2.6.1. At that time you can deliver this into the 2.6.z stream. Just need to wait for the 2.6.z stream to be open.

**Comment by GowthamShanmugam** [comment]
> @GowthamShanmugam Yes, so once 2.6 is GA @cbynum-redhat will open up the release-2.6 branch for 2.6.1. At that time you can deliver this into the 2.6.z stream. Just need to wait for the 2.6.z stream to be open.

Ack

**Comment by GowthamShanmugam** [comment]
Hi @fxiang1,
   Is ACM 2.6 already GA?

**Comment by KevinFCormier** [comment]
@GowthamShanmugam No. Planned date is September 7.

**Comment by GowthamShanmugam** [comment]
> @GowthamShanmugam No. Planned date is September 7.

@KevinFCormier @fxiang1 Since, ACM 2.6 is already GA, Kindly merge this PR for the Z-stream.

**Comment by KevinFCormier** [comment]
@GowthamShanmugam This PR is for `main` branch, so it will target 2.7. For 2.6.1, it will be a CVE only urgent fix, so please open a PR against `release-2.6` for that, but it is still not open for general deliveries.

**Comment by KevinFCormier** [comment]
@GowthamShanmugam Yesterday, 100% test coverage requirement was enabled for all ACM repos. `lint-check-unittests-sonarcloud` will not pass with 30.6% coverage.

**Review by fxiang1 — APPROVED** [review]
LGTM! I think they lowered the coverage requirement temporarily now

**Pull request merged**